### DISCUSSION TITLE
Variable text length Layout

### DISCUSSION THREAD

**Discussion by karsood** [Sunbird-Saral/Community#64] [Ideas]
Dear team, I am sharing below a requirement for Saral. Please do help with the creation of a layout.

Concept: 
Include scannable forms at the end of every chapter in each student's workbook where a teacher marks the performance of the student against pre-defined learning outcomes which are a part of the layout.
Each student receives one workbook. The teacher fills OMR for each student directly in the workbook.
There will be five columns, one for the learning outcome and 4 for four OMR bubbles against each learning outcome.

Challenges to be solved for: 
1. The length of each learning outcome differs and exists in 2 languages (Hindi and English)
- In Hindi, the expected maximum word length will be about 15 words (approx 75 characters) and the minimum would be 7 (approx 35 chars) words. Expected font size 14. 
- In English, the expected maximum word length will be about 20 words (approx 140 characters) and the minimum would be 8 (approx 60 chars) words. Expected font size 14. Expected font : Times New Roman
2. Number of learning outcomes ranges from 2 to 13 and the layout for each chapter should contain only as many number of rows as there are learning outcomes being tested for.
3. Leave out some space on one side so that the sheets can be bound into a workbook.
4. Enhancement in scanning ability in order to scan data filled in a form which is a part of a book.

Desired Solution: A layout with uniform width/length of rows + each chapter to have exactly the same number of rows as there are learning outcomes listed.

Potential Solution: 
- If such a layout can contain a maximum of say 5 rows then we can have multi-page scanning enabled with smoother operation than currently available to cater for combinations of 5+1, 5+2, 5+3, 5+4, 5+5, 5+5+1, 5+5+2, 5+5+3 learning outcomes.
- The app should ask teachers if there are additional pages to be scanned for these ROIs that are selected. This way the multi-page functionality automatically kicks in. At the end, collated data for all sheets can be seen in a single review page.

**Comment by dileep-gadiraju** [comment]
Hi @karsood, Can you give some draft drawings you used in the last discussion we had?  That will help initiate discussion and get inputs from UX designer also.

**Reply by karsood** [reply]
<img width="561" alt="image" src="https://user-images.githubusercontent.com/94509289/212854039-917b8a5d-f8a0-4838-9779-756b78d93d53.png">
<img width="674" alt="image" src="https://user-images.githubusercontent.com/94509289/212854097-eaab587a-a528-4b30-9d85-5d8a09a0f21c.png">
